### PR TITLE
JS2 week2 Homework

### PR DESCRIPTION
**Pull request by Sarulathaanbu**

### CODE REVIEW

File: javascript/javascript2/week2/movie.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  }
+  if (movie.rating < 4) {
+    return movie.tag = "Bad";
+  }
+}).map((movie) => movie.tag);
+
+console.log(rating);
+// Using chaining, first filter the movies array to only contain the movies rated higher than 6.
+// Now map the movies array to only the rating of the movies.
+const chaining = movies.filter((x) => x.rating > 6).map((x) => x.title);
```
Comment (obelitz) [inline javascript/javascript2/week2/movie.js:27]: This should only return the rating, so it should be:
```suggestion
const wellRatedMovies = movies.filter((movie) => movie.rating > 6).map((movie) => movie.rating);
```

File: javascript/javascript2/week2/movie.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+console.log(chaining);
+// Count the total number of movies containing any of following keywords: Surfer, Alien or Benjamin.
+//  So if there were 3 movies that contained Surfer, 1 with Alien and 2 with Benjamin, you would return 6.
+//   Can you make sure the search is case insensitive?
+
+const nameFilter = movies.filter((movie) =>
+  (movie.title.toLowerCase().indexOf("surfer") >= 0) ||
+  (movie.title.toLowerCase().indexOf("alien") >= 0) ||
+  (movie.title.toLowerCase().indexOf("benjamin") >= 0));
```
Comment (obelitz) [inline javascript/javascript2/week2/movie.js:36]: You can use `includes` here to make it simpler:
```js
  (movie.title.toLowerCase().includes("surfer")) ||
  (movie.title.toLowerCase().includes("alien")) ||
  (movie.title.toLowerCase().includes("benjamin")));
```

File: javascript/javascript2/week2/movie.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+console.log(nameFilter.length);
+
+// Calculate the average rating of all the movies using reduce.
+const movieRating = movies.map(movie => movie.rating);
+var averageRating = movieRating.reduce((total, rating) => total + rating) /
```
Comment (obelitz) [inline javascript/javascript2/week2/movie.js:41]: This should be a const. 
In general, you should avoid `var`. 
We usually only use `let` or `const`.

**Review by obelitz — COMMENTED** [review]
Good job! ✨ 

Notes:
- You should think more of using descriptive names for you variables. Eg. it it's an array of movies with ratings, it should be `moviesWithRatings`.
- I have added some ideas for optimization.

File: javascript/javascript2/week2/hyf-bay/main.js [hunk 4]
```diff
@@ -0,0 +1,90 @@
+console.log('Script loaded');
+
+const products = getAvailableProducts();
+const productsUl = document.querySelector('section.products ul');
+console.log(productsUl);
+
+function renderProducts(products) {
```
Comment (obelitz) [inline javascript/javascript2/week2/hyf-bay/main.js:7]: This has the same name as the const. 
That might be an issue.

File: javascript/javascript2/week2/movie.js (hunk 3 above)
Comment (Sarulathaanbu) [reply javascript/javascript2/week2/movie.js]: 👍🏻

File: javascript/javascript2/week2/movie.js (hunk 1 above)
Comment (Sarulathaanbu) [reply javascript/javascript2/week2/movie.js]: 👍🏻

File: javascript/javascript2/week2/movie.js [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+console.log(shortName);
+// Create an array of movie titles with long movie titles
+const longName = movies.filter((x) => x.title.length > 10).map((x) => x.title);
+console.log(longName);
+// Count the number of movies made between 1980-1989 (including both the years)
+const yearBet = movies.filter((x) => x.year >= 1980 && x.year <= 1989).map((x) =>
+  x.title);
+console.log(yearBet);
+// Create a new array that has an extra key called tag. The tag is based on the rating: Good (>= 7), Average (>= 4 and < 7), Bad (< 4)
+const rating = movies.filter(function(movie) {
```
[the remark this answers is not in the document]
Comment (Sarulathaanbu) [reply javascript/javascript2/week2/movie.js]: 👍🏻

File: javascript/javascript2/week2/movie.js [hunk 6]
```diff
@@ -0,0 +1,53 @@
+// Create an array of movies containing the movies with a short title (you define what short means)
+const shortName = movies.filter((x) => x.title.length < 10).map((x) => x.title);
+console.log(shortName);
+// Create an array of movie titles with long movie titles
+const longName = movies.filter((x) => x.title.length > 10).map((x) => x.title);
+console.log(longName);
+// Count the number of movies made between 1980-1989 (including both the years)
+const yearBet = movies.filter((x) => x.year >= 1980 && x.year <= 1989).map((x) =>
+  x.title);
```
[the remark this answers is not in the document]
Comment (Sarulathaanbu) [reply javascript/javascript2/week2/movie.js]: 👍🏻

File: javascript/javascript2/week2/movie.js [hunk 7]
```diff
@@ -0,0 +1,53 @@
+// Create an array of movies containing the movies with a short title (you define what short means)
+const shortName = movies.filter((x) => x.title.length < 10).map((x) => x.title);
```
[the remark this answers is not in the document]
Comment (Sarulathaanbu) [reply javascript/javascript2/week2/movie.js]: 👍🏻

File: javascript/javascript2/week2/index.html [hunk 8]
```diff
@@ -0,0 +1,12 @@
+<!DOCTYPE html>
+<html lang="en" dir="ltr">
+  <head>
+    <meta charset="utf-8">
+    <title></title>
+  </head>
+  <body>
+    <script src="movie.js"></script>
+    <script src="movies.js"></script>
```
[the remark this answers is not in the document]
Comment (Sarulathaanbu) [reply javascript/javascript2/week2/index.html:8]: 👍🏻

File: javascript/javascript2/week2/hyf-bay/main.js [hunk 9]
```diff
@@ -0,0 +1,90 @@
...
+const sortName = document.getElementById('sortname');
+
+
+//   Search Box /////////////////////
+userInput.addEventListener("input", function(event) {
+  const productName = event.target.value;
+  console.log(productName);
+  const filteredProducts = products.filter((product) =>
+    product.name.toLowerCase().includes(productName.toLowerCase())
+  );
```
[the remark this answers is not in the document]
Comment (Sarulathaanbu) [reply javascript/javascript2/week2/hyf-bay/main.js:41]: 🙂

File: javascript/javascript2/week2/hyf-bay/main.js [hunk 10]
```diff
@@ -0,0 +1,90 @@
...
+    li.innerHTML =
+      `
+            <ul>
+                <li>${product.name}</li>
+                <li>${product.price}</li>
+                <li>${product.rating}</li>
+                <ul class="ships-to">${shipsToHTML}</ul>
+            </ul>
+        `;
+    productsUl.appendChild(li);
```
[the remark this answers is not in the document]
Comment (Sarulathaanbu) [reply javascript/javascript2/week2/hyf-bay/main.js]: 🙂

File: javascript/javascript2/week2/hyf-bay/main.js [hunk 11]
```diff
@@ -0,0 +1,90 @@
+console.log('Script loaded');
+
+const products = getAvailableProducts();
+const productsUl = document.querySelector('section.products ul');
+console.log(productsUl);
+
+function renderProducts(products) {
+  products.forEach(product => {
+    const li = document.createElement('li');
```
[the remark this answers is not in the document]
Comment (Sarulathaanbu) [reply javascript/javascript2/week2/hyf-bay/main.js]: 👍🏻

File: javascript/javascript2/week2/hyf-bay/main.js [hunk 12]
```diff
@@ -0,0 +1,90 @@
...
+
+    let shipsToHTML = '';
+    product.shipsTo.forEach(country => shipsToHTML += `<li>${country}</li>`);
```
[the remark this answers is not in the document]
Comment (Sarulathaanbu) [reply javascript/javascript2/week2/hyf-bay/main.js:12]: 🙂

File: javascript/javascript2/week2/hyf-bay/main.js [hunk 13]
```diff
@@ -0,0 +1,90 @@
+console.log('Script loaded');
+
+const products = getAvailableProducts();
+const productsUl = document.querySelector('section.products ul');
```
[the remark this answers is not in the document]
Comment (Sarulathaanbu) [reply javascript/javascript2/week2/hyf-bay/main.js:4]: 🙂

File: javascript/javascript2/week2/double.js [hunk 14]
```diff
@@ -0,0 +1,15 @@
...
+
+for (let i = 0; i < numbers.length; i++) {
+  if (numbers[i] % 2 !== 0) {
+    newNumbers.push(numbers[i] * 2);
+  }
+}
+
+console.log("The doubled numbers are", newNumbers); // [2, 6]
+
+const oddNum = numbers.filter(num => num % 2);
```
[the remark this answers is not in the document]
Comment (Sarulathaanbu) [reply javascript/javascript2/week2/double.js]: 👍🏻

File: javascript/javascript2/week2/double.js [hunk 15]
```diff
@@ -0,0 +1,15 @@
+let numbers = [1, 2, 3, 4];
+let newNumbers = [];
```
[the remark this answers is not in the document]
Comment (Sarulathaanbu) [reply javascript/javascript2/week2/double.js]: 👍🏻

File: javascript/javascript2/week2/hyf-bay/main.js (hunk 4 above)
Comment (Sarulathaanbu) [reply javascript/javascript2/week2/hyf-bay/main.js:7]: 👍🏻

**Pull request merged**